server/battles/BattleResultProcessor.cpp

```diff
@@ -241,6 +241,11 @@ void BattleResultProcessor::endBattle(const CBattleInfoCallback & battle)
 			battleResult->exp[1] += 500;
 		if(heroDefender)
 			battleResult->exp[0] += 500;
+
+		// Give 500 exp to winner if a town was conquered during the battle
+		const auto * defendedTown = battle.battleGetDefendedTown();
+		if (defendedTown && battleResult->winner == BattleSide::ATTACKER)
+			battleResult->exp[BattleSide::ATTACKER] += 500;
```

Comment:
That's what the original issue report says, so I'd assume so:
Expected behavior
> Expected behavior
> Experience awarded for conquering a town equals 500+total hp of defeated army if the town had no hero
> Experience awarded for conquering a town equals 1000+total hp of defeated army if the town had a hero

I have yet to test this out myself with the original HoMM engine though, will need to check before merge.

Reply:
I'm not very expert of OH3, so maybe it's correct behavior, just want to pay attention on that

Reply:
Unfortunately I do not have the time to test this right now and the phrasing on HoMM wiki is very ambigous:
> If an enemy hero is defeated or an enemy town is captured during a siege, the hero receives 500 extra experience points.

I can probably test tomorrow night on OH3 but if someone else has time or knowledge to confirm that +1000xp is the original behavior when defeating a hero AND capturing a town then it would be appreciated.

Reply:
According to Warzyw in linked issue OH3 indeed gives 1000 exp in this case